### ISSUE TITLE
Computed column error when SetOutputIdentity is set to true during BulkInsert 

### ISSUE THREAD

**Issue by nimeshdhruve** [borisdj/EFCore.BulkExtensions#54]
Can you please confirm the following issue?

I noticed BulkInsert fails if you have computed column in table and you are using SetOutputIdentity = true.

Code for loading the data:
```
public IEnumerable<TestResult> SeedTestResults(IEnumerable<TestResult> data)
{
	///This code was auto generated, if you make any change in this method then next time when the code is regenerated your changes will be overwritten 
	if (data.Any() == false)
		return data;

	using (var transaction = Database.BeginTransaction())
	{
		var items = data.ToList();
                //No effect of PropertiesToExclude for Insert
		//List<string> excludedProperties = null;
		//excludedProperties = new List<string>();
		//excludedProperties.Add("TestStatus");

		this.BulkInsert(items, new BulkConfig { SetOutputIdentity = true, PreserveInsertOrder = true /*, PropertiesToExclude = excludedProperties*/ });
		var relatedItems = new List<TestMeasurement>();
		foreach (var item in items)
		{
			foreach (var subItem in item.TestMeasurements)
			{
				subItem.TestResultId = item.TestResultId;
			}
			relatedItems.AddRange(item.TestMeasurements);
		}
		this.BulkInsert(relatedItems, new BulkConfig { SetOutputIdentity = true, PreserveInsertOrder = true });
		transaction.Commit();
		return data;
	}	
}

```
Following is the merge query that gets generated:
```
MERGE dbo.[TestResults] WITH (HOLDLOCK) AS T USING (SELECT TOP 60 * FROM dbo.[TestResultsTemp4b97610a] ORDER BY [TestResultId]) AS S ON T.[TestResultId] = S.[TestResultId] WHEN NOT MATCHED THEN INSERT ([BuildCode], [CustomerSequenceNumber], [ErrorMessage], [JobId], [RecipeId], [ShiftId], [StationId], [TestDate], [TestStatus], [TotalTimeInSeconds]) VALUES (S.[BuildCode], S.[CustomerSequenceNumber], S.[ErrorMessage], S.[JobId], S.[RecipeId], S.[ShiftId], S.[StationId], S.[TestDate], S.[TestStatus], S.[TotalTimeInSeconds]) OUTPUT INSERTED.[TestResultId], INSERTED.[BuildCode], INSERTED.[CustomerSequenceNumber], INSERTED.[ErrorMessage], INSERTED.[JobId], INSERTED.[RecipeId], INSERTED.[ShiftId], INSERTED.[StationId], INSERTED.[TestDate], INSERTED.[TestStatus], INSERTED.[TotalTimeInSeconds] INTO dbo.[TestResultsTemp4b97610aOutput];
````
Table script
```
CREATE TABLE [dbo].[TestResults](
	[TestResultId] [int] IDENTITY(1,1) NOT NULL,
	[JobId] [nvarchar](50) NULL,
	[BuildCode] [nvarchar](50) NULL,
	[CustomerSequenceNumber] [nvarchar](50) NULL,
	[TestStatus]  AS (case when ltrim(rtrim(isnull([ERRORMESSAGE],'')))='' then 'Pass' else 'Fail' end) PERSISTED NOT NULL,
	[ErrorMessage] [nvarchar](512) NULL,
	[TestDate] [datetime] NOT NULL,
	[TotalTimeInSeconds] [decimal](18, 2) NOT NULL,
	[StationId] [int] NOT NULL,
	[ShiftId] [int] NOT NULL,
	[RecipeId] [int] NULL,
 CONSTRAINT [PK_TestResults] PRIMARY KEY CLUSTERED 
(
	[TestResultId] ASC
)WITH (PAD_INDEX = OFF, STATISTICS_NORECOMPUTE = OFF, IGNORE_DUP_KEY = OFF, ALLOW_ROW_LOCKS = ON, ALLOW_PAGE_LOCKS = ON) ON [PRIMARY]
) ON [PRIMARY]
GO
...
...
...
--removed indexes and other details for brevity

```
OnModelCreating code for TestResult table
```
modelBuilder.Entity<TestResult>(entity =>
{
	entity.Property(e => e.TestStatus)
		  .HasConversion<string>();

	entity.Property(e => e.ErrorMessage).HasMaxLength(512);

	entity.Property(e => e.JobId).HasMaxLength(50);

	entity.Property(e => e.TestDate).HasColumnType("datetime");

	entity.Property(e => e.TestStatus)
		.IsRequired()
		.HasMaxLength(4)
		.IsUnicode(false)
		.HasComputedColumnSql("(case when ltrim(rtrim(isnull([ERRORMESSAGE],'')))='' then 'Pass' else 'Fail' end)");

	entity.HasOne(d => d.Recipe)
		.WithMany(p => p.TestResults)
		.HasForeignKey(d => d.RecipeId)
		.HasConstraintName("FK_TestResults_Recipes");

	entity.HasOne(d => d.Shift)
		.WithMany(p => p.TestResults)
		.HasForeignKey(d => d.ShiftId)
		.OnDelete(DeleteBehavior.ClientSetNull)
		.HasConstraintName("FK_TestResults_PlantShifts");

	entity.HasOne(d => d.Station)
		.WithMany(p => p.TestResults)
		.HasForeignKey(d => d.StationId)
		.OnDelete(DeleteBehavior.ClientSetNull)
		.HasConstraintName("FK_TestResults_Stations");
});

```

**Comment by borisdj** [comment]
Yes I can.
The problem is in _QueryOutputTable_ method:
`context.Set<T>().FromSql<T>(SqlQueryBuilder.SelectFromOutputTable(this));`
where SQL does not Select CalculatedColumn but Entity expects it.
I will see to fix it soon.

**Comment by nimeshdhruve** [comment]
Thanks, I am inserting more than 9 million records (on a daily basis) and I really look forward to use BulkInsert. Regular EF Core takes more than 20mins even with doing things in Parallel.

**Comment by borisdj** [comment]
Fixed and NuGet 2.1.2 published.
Also _ComputedColumns_ are now automatically excluded so it's not required to use _PropertiesToExclude_ for their explicit exclusion.

**Comment by maxchu92** [comment]
Using 2.1.9 version.

Using Data Annotation, or FluentAPI does not work.

Data Annotation
`public class MyClass {`
`[DatabaseGenerated(DatabaseGeneratedOption.Identity)]`
`public MyProp { get; set; }`
`}`


FluentAPI
`modelBuilder.Entity<My_Table>().Property(p => p.MyProp).ValueGeneratedOnAddOrUpdate();`

**Comment by maxchu92** [comment]
Submitted as a new issue
